pkgs/by-name/st/styluslabs-write/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchpatch,
+  stdenv,
+  gitMinimal,
+  pkg-config,
+  SDL2,
+  xorg,
+  dbus,
+  libcap,
+  libGL,
+  roboto,
+  imagemagick,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "styluslabs-write";
+  version = "2024-09-26";
+
+  src = fetchFromGitHub {
+    owner = "styluslabs";
+    repo = "Write";
+    rev = "7110e5db87c154d28efd9507be4ab768af7403e0";
+    hash = "sha256-/IYI/trC14IBYtYn9LkjNrMZoCn6jtQIgq0ZqnsLISs=";
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    # Take SDL from buildInputs rather than vendor source
+    (fetchpatch {
+      url = "https://github.com/styluslabs/Write/pull/3/commits/11d1f9e96984bad69b531f40bb9efd9daada7fa4.patch";
+      hash = "sha256-q19JQm/eSmoyU8s4nY9B8XUdC5J7j9V6P6bQL/ukPG0=";
+    })
+  ];
+
+  preBuild = ''
+    # Bundled with a SDL2 fork to which it is statically linked.
+    # For nixpkgs, we apply a patch to dynamically link to SDL2 instead.
+    # TODO: We may still need to use this fork for non-Linux platforms.
+    rm -rf SDL/
+    pushd syncscribble
+  '';
+
+  postBuild = ''
+    popd
+  '';
+
+  strictDeps = true;
+
+  hardeningDisable = [ "format" ];
+  makeFlags = [
+    "DEBUG=0"
+    "GITREV=${finalAttrs.src.rev}"
+    "USE_SYSTEM_SDL=1"
+  ];
+
+  nativeBuildInputs = [
+    imagemagick # magick
+    pkg-config
+    gitMinimal
```

Comment:
https://github.com/styluslabs/Write/blob/7110e5db87c154d28efd9507be4ab768af7403e0/syncscribble/Makefile#L79-L84

I moved the git commit hash extraction to `postFetch` and removed `gitMinimal`.
Previously I was setting GITREV directly to `${finalAttrs.rev}`, but that will break in the future if a tagged release is created by upsteam.

Reply:
Well, the git processes wouldn't have a way to access the git state, so there's no point in having it.

What you could do here and what would be slightly preferable to the postFetch hack would be to have an updater script which produces a set of data which includes tagged version, hash rev and rev count which you'd simply import to use in the drv.

That's really a cherry on top thing though, this will work just fine.